skimage/transform/radon_transform.py

```diff
@@ -63,62 +75,75 @@ def radon(image, theta=None, circle=True, *, preserve_range=False):
 
     image = convert_to_float(image, preserve_range)
 
+    def_rot_center = _get_default_center(np.array(image.shape, dtype=np.float32))
+    if center is not None:
+        center = np.array(center, dtype=np.float32)
+        rel_rot_center = center - def_rot_center
+    else:
+        rel_rot_center = np.zeros_like(def_rot_center)
+
     if circle:
         shape_min = min(image.shape)
-        radius = shape_min // 2
-        img_shape = np.array(image.shape)
+        radius = _get_default_center(shape_min)
+
+        img_shape = np.array(image.shape, dtype=int)
         coords = np.array(np.ogrid[: image.shape[0], : image.shape[1]], dtype=object)
-        dist = ((coords - img_shape // 2) ** 2).sum(0)
+        dist = ((coords - (def_rot_center + rel_rot_center)) ** 2).sum(axis=0)
```

Comment:
The innermost parenthesis is basically `center` -- I believe it would be much more readable to have `center` instead of `(def_rot_center + rel_rot_center)`. You just have to define `center` in the (default) case where `center is None`.